### PR TITLE
chore: Extract paging related functions into standalone hook

### LINKED ISSUE #1621: feat: Making the list page paginator a standalone component
# Feature request

Making the list page paginator a standalone component.
Related #1584 
Currently, a paginator has been added to each listing page and it is hoped that this state paginator can be made into a component that can be easily called from each listing page, once it is completed we need to add test cases to it.

TODO:
- [ ] Making the list page paginator a standalone component
- [ ] Adding front-end E2E testing 

image

### PR DESCRIPTION
**Pull request by oil-oil**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Extract paging related functions into standalone hook

**Related issues**

resolve #1621

**Checklist:**

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: web/src/hooks/usePagination.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useEffect, useState } from 'react';
+import { useLocation } from 'umi';
+import { history } from 'umi';
```
Comment (Baoyuantop) [inline web/src/hooks/usePagination.ts:19]: should be merged

**Comment by oil-oil** [comment]
@guoqqqi @Baoyuantop Please help me review.😃

**Review by Baoyuantop — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Review by guoqqqi — APPROVED** [review]
LGTM

**Review by bzp2010 — APPROVED** [review]

**Comment by oil-oil** [comment]
@LiteSun PTAL.

**Review by LiteSun — APPROVED** [review]
LGTM

**Pull request merged**